File: Backend/src/main/java/com/example/backend/dto/ShoppingListResponseDto.java
package com.example.backend.dto;

import java.time.LocalDateTime;
import java.util.List;

public record ShoppingListResponseDto(
        Integer id,
        Integer userId,
        String name,
        LocalDateTime createdAt,
        Boolean isCompleted,
        List<ShoppingListItemResponseDto> items
) {
}

File: MyApplication/data/src/main/java/com/example/data/Recipes/datasource/remote/dto/ShoppingListResponseDto.kt
package com.example.data.Recipes.datasource.remote.dto 

data class ShoppingListResponseDto(
    val id: Int,
    val userId: Int,
    val name: String,
    val createdAt: String,
    val isCompleted: Boolean,
    val items: List<ShoppingListItemResponseDto>
)